clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java

```diff
@@ -4249,7 +4249,8 @@ public void testListOffsetsMaxTimestampUnsupportedSingleOffsetSpec() {
         final TopicPartition tp0 = new TopicPartition("foo", 0);
 
         try (AdminClientUnitTestEnv env = new AdminClientUnitTestEnv(cluster, AdminClientConfig.RETRIES_CONFIG, "2")) {
-            env.kafkaClient().setNodeApiVersions(NodeApiVersions.create());
+            env.kafkaClient().setNodeApiVersions(NodeApiVersions.create(
+                    ApiKeys.LIST_OFFSETS.id, (short) 0, (short) 6));
```

Comment:
It is a bit annoying that we have to keep calling `prepareUnsupportedVersionResponse` below even when we provide the versions. In the end, setting the versions in this case does not change much because `prepareUnsupportedVersionResponse` will make the request fail anyway. This is inherently due to the implementation of the `MockClient` which requires queued responses in order to process queued requests. It is fine to keep it as it in this patch but we should really rework the `MockClient`.